sdk/src/compressed_account.rs

```diff
@@ -351,51 +220,16 @@ where
     T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Discriminator,
 {
     input_account: T,
-    address_seed: Option<[u8; 32]>,
-    merkle_context: PackedMerkleContext,
-    merkle_tree_root_index: u16,
-    address_merkle_context: PackedAddressMerkleContext,
 }
 
 impl<T> LightCloseAccount<T>
 where
     T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Discriminator,
 {
-    pub fn try_from_slice(
-        v: &[u8],
-        merkle_context: &PackedMerkleContext,
-        merkle_tree_root_index: u16,
-        address_merkle_context: &PackedAddressMerkleContext,
-    ) -> Result<Self> {
-        let input_account = T::try_from_slice(v)?;
-
-        Ok(Self {
-            input_account,
-            address_seed: None,
-            merkle_context: *merkle_context,
-            merkle_tree_root_index,
-            address_merkle_context: *address_merkle_context,
-        })
-    }
-
-    pub fn set_address_seed(&mut self, address_seed: [u8; 32]) {
-        self.address_seed = Some(address_seed);
-    }
+    pub fn try_from_slice(compressed_account: &CompressedAccount) -> Result<Self> {
+        let input_account = T::try_from_slice(&compressed_account.data.as_ref().unwrap().data)?;
```

Comment:
Nice that we are only using the `compressed_account.data.as_ref().unwrap().data`!
Could we avoid sending the rest of the compressed account then and just build it onchain?